### PR TITLE
Reorg: Add a dummy "jetpack.php" to the monorepo, and update docs

### PR DESCRIPTION
**Pull request by anomiex**
#### Changes proposed in this Pull Request:

People used to the old way of doing things may try to check out the monorepo directly in their WordPress `wp-content/plugins` folder, or they may already be doing so. That will no longer work to install Jetpack.

To make the failure case a little more user-friendly, we add a dummy `jetpack.php` to the monorepo root so WordPress will find a plugin. All this plugin does is display an admin notice telling the user they're doing it wrong and pointing them to relevant documentation on how to do it right.

Said documentation needs to exist. So this PR also moves the existing Jetpack readme.md to the Jetpack plugin, and creates a new readme.md for the monorepo in the monorepo root. Once we get to the point of creating the built "Automattic/jetpack-production" mirror repo, the Jetpack-specific readme.md will show up there too.

Since both the Jetpack-specific readme.md and the Monorepo readme.md want to refer to a LICENSE.txt, we also need to copy that into the Jetpack plugin. Again, this will ensure the "Automattic/jetpack-production" mirror repo has LICENSE.txt too once we create it.

And since there's now a fairly significant duplication of content between the Monorepo and Jetpack in readme.md and LICENSE.txt, let's add a pre-commit and CI check to ensure those remain in sync in the future.

And while I was looking at it, I updated a few doc refs in other files too.

#### Jetpack product discussion

1199714989172614-as-1199714989172649

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Try putting the monorepo into the WordPress plugins directory. With the docker env, you can do that like
    * Set up the docker env as usual.
    * `yarn docker:sh`, then `ln -s /usr/local/src/jetpack-monorepo /var/www/html/wp-content/plugins/jetpack-monorepo`
* Go into the WP admin and enable the "Jetpack Monorepo (not a real plugin)" plugin.
* See the admin notice. The link included won't link to the right place of course.

* View the changes to the docs, see that they look appropriate. In particular, look at [the Monorepo readme](https://github.com/Automattic/jetpack/blob/reorg/add-dummy-jetpack.php/readme.md) and [the Jetpack plugin readme](https://github.com/Automattic/jetpack/blob/reorg/add-dummy-jetpack.php/projects/plugins/jetpack/readme.md).

#### Proposed changelog entry for your changes:

* None needed, this will be subsumed by the reorg.

### CODE REVIEW

File: tools/check-copied-files.sh [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+		else
+			echo "$MSG"
+		fi
+
+		diff -u /dev/fd/3 --label "$1" /dev/fd/4 --label "$2" 3<<<"$A" 4<<<"$B" || true
+		EXIT=1
+	fi
+}
+
+compare readme.md projects/plugins/jetpack/readme.md '^## Security'
```
Comment (kraftbj) [inline tools/check-copied-files.sh:73]: As I understand it, this means that everything after `## Security` will be checked and kept in sync, as it lacks the last parameter.

Looking at the readme file, I'm okay if we don't keep `Team` in sync. We can probably remove that from the jetpack-production one IMO, only keeping Security and License in sync.

File: projects/plugins/jetpack/readme.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+Contributions have been and continue to be made by dozens of other Automatticians, like:
+
+@aldavigdis, @allendav, @apeatling, @azaozz, @bazza, @beaulebens, @briancolinger, @cfinke, @daniloercoli, @chaosexanima, @ebinnion, @enejb, @eoigal, @georgestephanis, @gibrown, @gititon, @gravityrail, @jasmussen, @jblz, @jeffgolenski, @jeherve, @jessefriedman, @joanrho, @justinshreve, @keoshi, @koke, @kovshenin, @kraftbj, @lancewillett, @lezama, @martinremy, @mdawaffe, @MichaelArestad, @mtias, @mcsf, @mdawaffe, @nickmomrik, @obenland, @oskosk, @pento, @roccotripaldi, @stephdau, @Viper007Bond, @xyu and @yoavf.
+
+Our _awesome_ happiness engineers are @a8ck3n, @aheckler, @almoyen, @annezazuu, @bikedorkjon, @cena, @chaselivingston, @chickenn00dle, @coder-karen, @codestor4, @csonnek, @danjjohnson, @dericleeyy, @gaurav1984, @gsmumbo, @htdat, @jenhooks, @jeremypaavola, @joeboydston, @joendotcom, @lizthefair, @macmanx2, @madhattermattic, @mbhthompson, @mzakariya, @NujabesSoul, @ntpixels, @pmciano, @rachelsquirrel, @rcowles, @richardmtl, @snowmads, @stefmattana, and @tmmbecker.
```
Comment (kraftbj) [inline projects/plugins/jetpack/readme.md:54]: We should handle this in a follow-up PR, but a lot of these names aren't accurate anymore :-/

**Review by kraftbj — CHANGES_REQUESTED** [review]
It all looks good, but a request to remove Teams from the data kept in sync between the two readmes.

**Comment by kraftbj** [comment]
Separately, I created https://github.com/Automattic/jetpack-production so we would have that reserved for when we get to that step.

File: tools/check-copied-files.sh (hunk 1 above)
Comment (anomiex) [reply tools/check-copied-files.sh:73]: To be clear, you're asking to delete the "Team" section entirely from projects/plugins/jetpack/readme.md?

Comment (kraftbj) [reply tools/check-copied-files.sh:73]: Yes. IMO, having it at the root-level only is sufficient.

**Review by kraftbj — APPROVED** [review]

**Pull request merged**